services/py-api/Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM python:3.12.6-bullseye AS builder
+LABEL authors="ivanobreshkov"
+
+# https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0
+
+RUN pip install poetry
+
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
+
+WORKDIR /app
+
+COPY pyproject.toml poetry.lock README.md ./
+
+RUN poetry install --no-root && rm -rf $POETRY_CACHE_DIR
+
+# The runtime image, used to just run the code provided its virtual environment
+FROM python:3.12.6-slim-bullseye AS runtime
+
+RUN apt-get update && apt-get install -y --no-install-recommends curl && rm -rf /var/lib/apt/lists/*
+
+ENV VIRTUAL_ENV=/app/.venv \
+    PATH="/app/.venv/bin:$PATH"
+
+COPY --from=builder ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+
+WORKDIR /app
+
+COPY src ./src
+COPY .env ./
+
+EXPOSE 8080
+
+# As we are using self-signed certs we don't check the validity of the certs with -k
+HEALTHCHECK --interval=1m --timeout=10s CMD curl -k -f https://localhost:6969/email-service/v1/ping
```

Comment:
This might be out of scope but what I would suggest you do here is:

- Have the domain as an environment variable so that production containers run the check against the proper url and catch ssl issues.
- If you're opting into running self-signed certs, install them into the container so that you don't have to skip verification.

Reply:
I like the idea with the env variable for the domain. Apparently i have forgotten to copy the certs folder. Nevertheless, as this will be running only on DEV and PROD environmentsI don't think we need the self signed certs at all. What do you think? Because if we do have them the healthcheck will have to choose to verify the certs or not to verify them, depending on the domain. If it's localhost we don't verify them, if it's not we should verify them.

I am a bit concerned if the check towards https://thehub-aubg.com/api/v3/ping will work, as last time I did that I had some issues with it. Have to try it.

Reply:
I think the idea with self-signed certs is good. Something which was an issue with the previous configuration was that it was incredibly hard to debug in local environments. You couldn't just spin up everything from the get go and verify if requests are propagated correctly between containers.